helper/cert/source_gen.go

```diff
@@ -393,3 +445,41 @@ func parseCert(pemValue []byte) (*x509.Certificate, error) {
 
 	return x509.ParseCertificate(block.Bytes)
 }
+
+// decodeCerts decodes a caBundle ([]byte) into a list of certs ([][]byte)
+func decodeCerts(caBundle []byte) tls.Certificate {
+	certs := tls.Certificate{}
+	remainder := caBundle
+	var next *pem.Block
+	for {
+		next, remainder = pem.Decode(remainder)
+		if next == nil {
+			break
+		}
+		if next.Type == "CERTIFICATE" {
```

Comment:
What might we throw away in the else clause, we would never expect anything right? Perhaps worth logging a warning to surface anything unexpected?